polyply/tests/test_top_parser.py

```diff
@@ -161,6 +157,31 @@ def test_ifdefs(def_statements, bonds):
         print(top.force_field.blocks["GLY"].interactions['bonds'])
         assert top.force_field.blocks["GLY"].interactions['bonds'] == bonds
 
+    @staticmethod
+    @pytest.mark.parametrize('def_statements',
+          ("""
+          #ifdef random
+          #include "random.itp"
+          #endif
+          """,
+          """
+          #define random
+          #ifndef random
+          #include "random.itp"
+          #endif
+          """))
+    def test_ifdefs_files(def_statements):
+        """
+        test files in ifdef/ifndef statements are only
+        read if a define has been read before
+        """
+        new_lines = textwrap.dedent(def_statements)
+        new_lines = new_lines.splitlines()
+        force_field = vermouth.forcefield.ForceField(name='test_ff')
+        top = Topology(force_field, name="test")
+        polyply.src.top_parser.read_topology(new_lines, top)
+        assert True
```

Comment:
Can't we do something better?

Reply:
In what sense better? All it is supposed to do is do nothing meaning it doesn't fail. I can't imagine really a more through test for this.

Reply:
Make sure random.itp exists, and that the things it contains do not end up in the FF. On the other hand, maybe it's good that random.itp doesn't exist. That way you can be sure parsing isn't even tried.

Better to leave it as is.